src/app/api/chat/create/route.ts

```diff
@@ -0,0 +1,66 @@
+//api/chat/create/
+//The chat will be created and the offer table should be updated with the chatid
+import { prisma } from "@/src/app/lib/db";
+import { NextResponse } from "next/server";
+import { v4 as uuidv4 } from 'uuid'; // For generating a unique chatId
+
+/**
+ * Create a chat for gig offer
+ * @param req { offer_id: number }
+ * @returns Success or error response
+ * @author horlarmmy
+ */
+
+export async function POST(req: Request) {
+  try {
+    // Extract parameters from the request
+    const body = await req.json();
+    const { offerId } = body;
+
+    if (!offerId) {
+      return NextResponse.json({ message: 'Invalid data. Offer ID is required' }, { status: 400 });
+    }
+
+    // Check if the offer exists
+    const offer = await prisma.gigOffer.findUnique({
+      where: { offerId: Number(offerId) },
+    });
+
+    if (!offer) {
+      return NextResponse.json({ message: 'Offer not found' }, { status: 404 });
+    }
+
+    if (offer.chatId) {
+        return NextResponse.json({ message: 'Chat already created!!!' }, { status: 404 });
+      }
+
+    // Generate a unique chatId
```

Comment:
Currently I was using chat id as a pattern of ClientId-FreelancerId-GigId. We need to discuss on the approach to use chatId.

Reply:
I don't understand what you mean